service/qos/service_level_controller.cc

```diff
@@ -89,12 +90,21 @@ void service_level_controller::set_distributed_data_accessor(service_level_distr
 future<> service_level_controller::stop() {
     // unregister from the service level distributed data accessor.
     _sl_data_accessor = nullptr;
+    try {
+        co_await std::exchange(_distributed_data_updater, make_ready_future<>());
+    } catch (...) {
+        // ignore all exceptions from distributed updater initialization
+    }
     if (this_shard_id() == global_controller) {
         // abort the loop of the distributed data checking if it is running
         _global_controller_db->dist_data_update_aborter.request_abort();
         _global_controller_db->notifications_serializer.broken();
+        try {
+            co_await std::exchange(_global_controller_db->distributed_data_update, make_ready_future<>());
+        } catch (const broken_semaphore& ignored) {
+        } catch (const sleep_aborted& ignored) {
+        }
```

Comment:
There's no point in having stop() functions pass an exception. So better to catch ... and log it here rather than confusing the caller.

Reply:
Yeah, but unexpected exceptions in stop() are caught in main.cc and printed - here I just explicitly ignore the boring ones.

Reply:
You can still catch the unexpected ones and log an error or warning about them.
An ERROR would help catch it in dtest and decide what do about it since we fail all dtests on any unexpected ERROR that is not explicitly marked to be ignored by the test, while it won't cause a coredump when shutting down in production.